docs/one-path-reachability-proofs.md

```diff
@@ -0,0 +1,174 @@
+Using a rule to advance a one-path reachability proof
+=====================================================
+
+This document describes how to transform a given a one-step reachability goal
+by advancing it using a given rewrite axiom.
+
+This document *does not describe* how to select an axiom to be used for
+advancing the goal.
+
+
+Problem statement
+-----------------
+
+Consider the following "one-path" reachability goal
+
+```
+∀ X . φ(X) → ◆ ∃ Y . ψ(X, Y)
+```
+where `X` and `Y` are sets of variables and ◆ denotes "strong-eventually"
+
+Consider also the axiom
+
+`∀ Z . α(Z) → ⚪ β(Z)`.
+
+where ⚪  denotes "strong-next".
+
+The steps below show how the given goal is transformed by "applying" the 
+given axiom.
+
+Unrolling "strong-eventually"
+------------------------------
+
+Let `Δ = ν f . ⚪ f` denote the formula for diverging computation.
+Then,  for any formula ψ:
+
+- `◆ ψ = ◇ ψ ∨ Δ` by the definition of "strong eventually"
+- `    = ψ ∨ ⚪ ◇ ψ ∨ Δ` by unrolling `◆`
+- `    = ψ ∨ ⚪ ◇ ψ ∨ ⚪ Δ` by unrolling `Δ`
+- `    = ψ ∨ ⚪ (◇ ψ ∨ Δ)`
+  because `⚪` commutes with `∨` 
+- `    = ψ ∨ ⚪ ◆ ψ`
```

Comment:
I would prefer an argument that avoids the non-constructive case split `◆ ψ = ◇ ψ ∨ Δ` if you have one.